lib/perl/Genome/Site/TGI/Command/ImportDataFromLims.pm

```diff
@@ -73,20 +73,31 @@ sub _process_instrument_data {
 
 
     try {
-        my ($bam_file, $lims_source_dir) = File::Basename::fileparse($lims_path);
+        my ($bam_file, $lims_source_dir);
+        if ($lims_path =~ /\.bam$/) {
+            ($bam_file, $lims_source_dir) = File::Basename::fileparse($lims_path);
+        } elsif (-d $lims_path) {
```

Comment:
Should we check not only that this is a directory, but that the files in there are fastq and in the correct format?

Reply:
Maybe!  I'd say the current version of our goal with this is to "accept whatever they've generated".  If it's not a BAM (or a .tar.gz of FASTQs) then none of our other code knows what to do with it anyway.  I've thought about adding better support for the current FASTQ format that we're likely to get, but it might not be worth plugging into all our legacy methods in `Genome::InstrumentData::Solexa`.